cmd/contact-auditor/main.go

```diff
@@ -0,0 +1,203 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/policy"
+)
+
+var queryInterrupted bool
+
+type contactAuditor struct {
+	db     *sql.DB
+	logger blog.Logger
+	clk    clock.Clock
+	grace  time.Duration
+}
+
+type result struct {
+	// Receiver for the `id` column.
+	ID int64
+
+	// Receiver for the `contact` column.
+	Contact  []byte
+	contacts []string
+}
+
+func (r *result) unmarshalContact() error {
+	var contact []string
+	err := json.Unmarshal(r.Contact, &contact)
+	if err != nil {
+		return err
+	}
+	r.contacts = append(r.contacts, contact...)
+	return nil
+}
+
+func (r *result) validateContact() error {
+	// Setup a buffer to store any validation problems we encounter.
+	var probsBuff strings.Builder
+
+	// Helper to write validation problems to our buffer.
+	writeProb := func(contact string, prob string) {
+		// Add validation problem to buffer.
+		probsBuff.WriteString(fmt.Sprintf("[ %q: %q ] ", contact, prob))
+	}
+
+	for _, contact := range r.contacts {
+		if strings.HasPrefix(contact, "mailto:") {
+			err := policy.ValidEmail(strings.TrimPrefix(contact, "mailto:"))
+			if err != nil {
+				writeProb(contact, err.Error())
+			}
+		} else {
+			writeProb(contact, "missing 'mailto:' prefix")
+		}
+	}
+
+	if probsBuff.String() != "" {
```

Comment:
check `probsBuff.Len() != 0`, so that you don't have to build out the whole string twice.